src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.netfx.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Instrument{T} is the base class from which all non-observable instruments will inherit from.
+    /// </summary>
+    /// <remarks>
+    /// This class supports only the following generic parameter types: <see cref="byte" />, <see cref="short" />, <see cref="int" />, <see cref="long" />, <see cref="float" />, <see cref="double" />, and <see cref="decimal" />
+    /// </remarks>
+#if ALLOW_PARTIALLY_TRUSTED_CALLERS
+        [System.Security.SecuritySafeCriticalAttribute]
+#endif
+    public abstract partial class Instrument<T> : Instrument where T : struct
+    {
+        [ThreadStatic] private KeyValuePair<string, object?>[] ts_tags = new KeyValuePair<string, object?>[3];
```

Comment:
Initializing ThreadStatics like this does not work. The initialization will only run on the first thread.

Reply:
Noah has suggested a different implementation https://github.com/dotnet/runtime/pull/52685#discussion_r632420307 which will check and initialize. Thanks.